llvm-passes/ObfuscateVar/ObfuscateVar.cpp

```diff
@@ -0,0 +1,210 @@
+    Value *varPointer = S.getPointerOperand();
+      
+    //dbgs() << "Store Inst :\n\t" << S << "\n\tValue is : " << *C0 <<  "\n\Address is : " << *varPointer << "\n\n";
+
+    AllocaInst* alloX1 = new AllocaInst(IntermediaryType, 0,"x1");
+    AllocaInst* alloX2 = new AllocaInst(IntermediaryType, 0,"x2");
+
+    BB->getInstList().insert(S, alloX1); //%x1 = alloca i32, align 4
+    BB->getInstList().insert(S, alloX2); //%x2 = alloca i32, align 4
+
+    //Value* valueOperand = S.getValueOperand();
+    //dbgs() << "Type: " << *valueOperand->getType() << "\n\n";
+
+    // Initialise x1 et x2 avec avec la "valeur" de la variable a spliter.
+    StoreInst *storeX1 = new StoreInst(S.getValueOperand(),alloX1,isVolatile,4);
+    StoreInst *storeX2 = new StoreInst(S.getValueOperand(),alloX2,isVolatile,4);
+    BB->getInstList().insert(S, storeX1); //store VAR, i32* %x1, align 4
+    BB->getInstList().insert(S, storeX2); //store VAR, i32* %x2, align 4
+
+    //Transfert x1,x2 dans des registres pour y effectuer des operations
+    LoadInst* loadX1 = new LoadInst(alloX1,"loadX1",isVolatile); 
+    LoadInst* loadX2 = new LoadInst(alloX2,"loadX2",isVolatile); 
+    BB->getInstList().insert(S, loadX1); //%xx = load i32* %x1
+    BB->getInstList().insert(S, loadX2); //%yy = load i32* %x2
+
+    // x1 et x2 sont mis a jour avec les bonnes valeurs.
+    // x1 = x1%10
+    // x2 = (int)x2/10
+    BinaryOperator* opRem = BinaryOperator::Create(Instruction::URem, loadX1, ConstantInt::get(IntermediaryType,10,false));
+    BinaryOperator* opDiv = BinaryOperator::Create(Instruction::UDiv, loadX2, ConstantInt::get(IntermediaryType,10,false));
+      
+    BB->getInstList().insert(S, opRem); //x1%10
+    BB->getInstList().insert(S, opDiv); //x2/10
+
+    // Sauvegarde les valeurs de x1,x2 precedement calculées
+    storeX1 = new StoreInst(opRem,alloX1,isVolatile,4);
+    storeX2 = new StoreInst(opDiv,alloX2,isVolatile,4);
+    BB->getInstList().insert(S, storeX1);
+    BB->getInstList().insert(S, storeX2);
+
+    // Enregistre les pointeurs de x1,x2 associé a la variable à diviser.
+    varsRegister.insert(std::pair<Value*,typeMapValue >(varPointer,std::make_pair(alloX1,alloX2)));
+ 
+  }
+  
+  Value* replaceAddOp(Instruction &Inst){
+    bool isVolatile = false;
+    Type *IntermediaryType = IntegerType::get(Inst.getParent()->getContext(),sizeof(typeInter)*8);//32bits
+    BinaryOperator *Binop = dyn_cast<BinaryOperator>(&Inst);
+    ConstantInt *C = dyn_cast<ConstantInt>(Binop->getOperand(1));
+    
+    Value *var = Binop->getOperand(0); //adresse de la variable
+
+    typeMap::iterator it; // Iterateur sur la map
+    
+    LoadInst *load = dyn_cast<LoadInst>(&*var);
+    Value* varPointer = load->getPointerOperand(); // Pointeur sur la variable de l'operande
+    
+    //dbgs() << Inst << "\n";
+    //dbgs() <<"Load instruction :"<<*load<<" Name is : "<<*load->getPointerOperand()<<"\n";
+
+    if((it=varsRegister.find(varPointer)) == varsRegister.end()){
```

Comment:
ok, I see and issue there. Your transformation pattern is:
1. register all store
2. whenever I have an addition, check if one of the argument is loading from one of this store and perform the conversion.

This is a really small subset of the possible obfuscation candidates. Here is what I propose:

1. starting from the beginning of the basic block, look for additions
2. whenever you find one, check each operand
3. if the operand is not already splitted, split it
4. once all operands are splitted, split your operation 
5. if the instruction is not an add and one of its operand uses a splitted value, merge the value back

something like this, I hope I'm clear...